solidity/contracts/converter/types/liquidity-pool-v1/LiquidityPoolV1Converter.sol

```diff
@@ -179,11 +181,14 @@ contract LiquidityPoolV1Converter is LiquidityPoolConverter {
         // sync the reserve balances
         syncReserveBalance(_sourceToken);
 
-        // calculate protocol fee from the conversion fee only. So it would take x% of conversion fee as the protocol fee.
-        // Also we don't need to substract anything from the return amount.
-        uint256 calculatedProtocolFee = calculateProtocolFee(_targetToken, fee);
+        /// @dev Protocol fee is just part of the fee above.
+        uint256 calculatedProtocolFee = calculateProtocolFee(fee);
+
+        /// Add the total helds token fee.
+        protocolFeeTokensHeld[_targetToken] = protocolFeeTokensHeld[_targetToken].add(calculatedProtocolFee);
+
+        /// Reserve balance needs to be substracted by the protocol fee.

```

Comment:
in order to match the code as well as english arithmetic syntax, the comment should rather say:
Reserve balance needs to be reduced by the protocol fee amount and base amount.
or even clearer:
Protocol fee amount and base amount need to be subtracted from the reserve balance.

Reply:
are we sure this is the correct way of doing things btw?
There's a call to syncReserveBalance a few lines above, and I wonder what this does exactly, as intuitively it seems it should already be doing what L191 is doing.
I'm having trouble locating that function's current version. Any help appreciated @cwsnt 
Will get back to it on second pass later today. Kinda saturating atm.

Reply:
```
function syncReserveBalance(IERC20Token _reserveToken) internal validReserve(_reserveToken) {
		if (_reserveToken == ETH_RESERVE_ADDRESS) reserves[_reserveToken].balance = address(this).balance;
		else reserves[_reserveToken].balance = _reserveToken.balanceOf(this).sub(protocolFeeTokensHeld[_reserveToken]);
	}
```


If syncReserveBalance would be called after protocolFeeTokensHeld is updated and after the amount was transfered, this line would not be necessary.
I'm quite sure that it's called before the token transfer on purpose to finish all state updates before calling an external contract.

I would appreciate a double check on .sub(calculatedProtocolFee) however. 
I reviewed the logic myself and think it's correct, because the amount excludes the fees (which stay on the converter), but the protocol fee which is on the converter should not be considered as part of the converters reserve. But like I said, double check here is appreciated :-)

Reply:
I have double-checked this `.sub(calculatedProtocolFee)` , and couldn't find any incorrect logic here.

However, as reference for you guys, i posted the simple scenario here.
Ex:
Initial Balance Pool = 1000 SOV
Total swapped = 200 SOV
Conversion Fee = 200 * 0.3% = 0.6 SOV
Protocol Fee = 0.05 / 0.3 * 0.3 = 0.1 SOV
Net Conversion Fee = 0.6 - 0.1 = 0.5 SOV
Total transferred to user = 200 - 0.6 = 199.4 SOV
Balance Pool = 1000 - 199.4 = 800.6 SOV
In this case reserve balance should be = 800.6 -0.1 = 800.5 SOV

Note: This 199.4 as a total transferred token, excludes the fee. Meanwhile, converter's reserves balance must spare some amount as the protocol fee. So we need to substract it with the protocolFee (0.1).